script/make_release.py

```diff
@@ -0,0 +1,225 @@
+        "type": "confirm",
+        "name": "confirmed",
+        "message": message,
+        "default": True
+    }]
+    confirmation = PyInquirer.prompt(confirm)["confirmed"]
+
+    if confirmation:
+        print(printMessage)
+    else:
+        print("Aborting.")
+        quit()
+
+
+def getLatestChangelogEntry(path):
+    release_post_content = ""
+    with open(path, "r", encoding="utf-8") as fp:
+        line_number = 0
+        section = None
+        for line in fp:
+            if line_number > 20:
+                if "## [" in line:
+                    break
+
+                if "###" in line and section != None:
+                    if len(section.split("\n")) > 3:
+                        release_post_content = release_post_content + section
+                    section = None
+
+                if section != None:
+                    section = section + line
+
+                if "###" in line and section == None:
+                    section = line
+
+            line_number = line_number + 1
+    return release_post_content
+
+
+def getReleasePost(version, path):
+    release_post_header = f"---\ncategories:\n\t- Release\ntags:\n\t- gh-pages\n\ntitle: {version}\n---\n\n"
+    release_post_headline = "{{page.title}} is live and ready for [download](https://github.com/MaibornWolff/codecharta/releases/tag/{{page.title}}). This version brings the following:\n\n"
+    release_post_content = getLatestChangelogEntry(path)
+    return release_post_header + release_post_headline + release_post_content
+
+
+# Check if we're on project root folder
+if not isRootFolder:
+    print("Please execute this script from the project root folder. Aborting.")
+    quit()
+else:
+    repo = git.Repo(root)
+
+
+# Check if there are any uncommited changes
+if repo.is_dirty():
+    print("Please commit your changes first and/or ignore untracked files in git. Aborting.")
+    quit()
+
+
+# Check if we are on master branch
+if repo.head != "master":
+    print("You can only release on master branch. Aborting.")
+    quit()
+
+
+# Get latest tag
+latest_tag = str(
+    sorted(repo.tags, key=lambda t: t.commit.committed_datetime)[-1])
+print(f"Last version tag in git is {latest_tag}")
+
+
+# Precalc new versions
+versions = latest_tag.split(".")
+
+major = int(versions[0])
+minor = int(versions[1])
+patch = int(versions[2])
+
+new_major_version = f"{major + 1}.0.0"
+new_minor_version = f"{major}.{minor + 1}.0"
+new_patch_version = f"{major}.{minor}.{patch + 1}"
+
+
+# Get release type and version
+questions = [{
+    "type": "list",
+    "name": "version",
+    "message": f"Do you want to release a major version [{new_major_version}], minor version [{new_minor_version}] or a patch [{new_patch_version}]?",
+    "choices": ["Major", "Minor", "Patch"],
+}]
+
+release_type = PyInquirer.prompt(questions)["version"]
+
+if release_type == None:
+    quit()
+if release_type == "Major":
+    new_version = new_major_version
+if release_type == "Minor":
```

Comment:
Ideally, these are all `if ... else if ... else ...`. Otherwise the CPU still has to do the computations.

In this specific case it would also be good to have the else case be a warning and `quit()` as we likely do not want to accept values other than `major`, `minor` and `patch`?

Reply:
That is true, but there are not other value selectable in the prompt.

Reply:
Awesome. In that case the only change I would still do is to use `elif`. Mainly to keep it in mind, since this is absolutely no hot code path ;-)